homeassistant/components/heos/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""Denon HEOS Media Player."""
+
+import asyncio
+import logging
+import voluptuous as vol
+
+from homeassistant.components.media_player.const import (
+    DOMAIN as MEDIA_PLAYER_DOMAIN)
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+DOMAIN = 'heos'
+REQUIREMENTS = ['aioheos==0.4.0']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Set up the HEOS component."""
+    from aioheos import AioHeosController
+
+    host = config[DOMAIN][CONF_HOST]
+    controller = AioHeosController(hass.loop, host)
+
+    try:
+        await asyncio.wait_for(controller.connect(), timeout=5.0)
```

Comment:
We use async_timeout in the core, so that's available as a replacement for `asyncio.wait_for`. I would recommend using the former instead, but it's not required.

https://github.com/aio-libs/async-timeout

Reply:
async_timeout has more appealing syntax, but I want to remove this later anyway and restructuring the connect function.